organizer-network/organizer.network
# bin/digest.js

This is the second stage of an incremental refactor process (see also: https://github.com/organizer-network/organizer.network/pull/25). The goal with this one was to remove the `/api/digest` API endpoint and replace it with something that can be invoked from a terminal command.

This PR begins to move database-querying functions like `get_person` and `get_context` into a new library `db.js`. It also adds a library for sending emails and such, `notify.js`.

I am still working out how to create proper tests for database- and email-related code. Perhaps I should look into mocking libraries. I'm looking at adding some of the libraries described in this article:
https://hackernoon.com/testing-node-js-in-2018-10a04dd77391

For now though, you should be able to test using this manual method:
* run the server with `npm start`
* set up two user accounts in the same group
* set one of the user accounts to receive digest emails
* send a message from the other account to queue up a digest message
* run `./bin/digest.js` to send the message

Also, the testing script process `npm test` should now exit properly upon completion.
